dashboard/app/helpers/javalab_files_helper.rb

```diff
@@ -53,7 +53,8 @@ def self.get_project_files(channel_id, level_id)
   # If the level doesn't have validation and/or a maze, those fields will not be present.
   def self.get_project_files_with_override_sources(sources, level_id)
     all_files = get_level_files(level_id)
-    all_files["sources"]["main.json"]["source"] = sources
+    all_files["sources"]["main.json"] = {}
+    all_files["sources"]["main.json"] = {source: sources}.to_json
```

Comment:
Why the double assignment here?

Reply:
I think I just missed cleaning this up, will update